askama_shared/src/input.rs

```diff
@@ -175,6 +176,35 @@ impl TemplateInput<'_> {
             }
         }
 
+        let mime_type = ext
+            .as_ref()
+            .and_then(|ext| mime_guess::from_ext(ext).first())
+            .or_else(|| mime_guess::from_path(&path).first())
+            .map(|mime| {
+                const TEXT_TYPES: [(mime::Mime, mime::Mime); 6] = [
+                    (mime::TEXT_PLAIN, mime::TEXT_PLAIN_UTF_8),
+                    (mime::TEXT_HTML, mime::TEXT_HTML_UTF_8),
+                    (mime::TEXT_CSS, mime::TEXT_CSS_UTF_8),
+                    (mime::TEXT_CSV, mime::TEXT_CSV_UTF_8),
+                    (
+                        mime::TEXT_TAB_SEPARATED_VALUES,
+                        mime::TEXT_TAB_SEPARATED_VALUES_UTF_8,
+                    ),
+                    (
+                        mime::APPLICATION_JAVASCRIPT,
+                        mime::APPLICATION_JAVASCRIPT_UTF_8,
+                    ),
+                ];
```

Comment:
I'd prefer to outline the const here, just dump it near the bottom of the module. Also, could there be a separate commit that moves the logic from `askama::mime` into askama_shared, forwarding the methods from the old place such that we don't duplicate the logic, and also a separate commit that changes the logic to what you have here?

Reply:
Done